python/ray/_private/runtime_env/pip.py

```diff
@@ -128,52 +128,65 @@ def _get_ray_version_and_path() -> Tuple[str, str]:
     @classmethod
     def _create_or_get_virtualenv(cls, path: str, cwd: str, logger: logging.Logger):
         """Create or get a virtualenv from path."""
+
+        python = sys.executable
+        virtualenv_path = os.path.join(path, "virtualenv")
+        virtualenv_app_data_path = os.path.join(path, "virtualenv_app_data")
+        current_python_dir = os.path.abspath(
+            os.path.join(os.path.dirname(python), "..")
+        )
+
         if cls._is_in_virtualenv():
-            # TODO(fyrestone): Handle create virtualenv from virtualenv.
+            # virtualenv-clone homepage:
```

Comment:
Could you please add a comment about why we need `virtualenv-clone` in this case?  It's clear from the PR description and the previous PR, but it would be good to have it as a comment in the code for easier reference.

Reply:
> 



> Could you please add a comment about why we need `virtualenv-clone` in this case? It's clear from the PR description and the previous PR, but it would be good to have it as a comment in the code for easier reference.

[`virtualenv-clone README`]() have some reasons why we shouldn't use virtualenv to relocate virtualenv:
```
Virtualenv provides a way to make virtualenv's relocatable which could then be copied as we wanted. 
However making a virtualenv relocatable this way breaks the no-site-packages isolation of the virtualenv 
as well as other aspects that come with relative paths and /usr/bin/env shebangs that may be undesirable.

Also, the .pth and .egg-link rewriting doesn't seem to work as intended. This attempts to overcome these 
issues and provide a way to easily clone an existing virtualenv.
```

Example: If we don't rewrite `.pth` file in `new_venv`, and this file directory to `old_venv  sit-package's dir`, then the package A  which in `old_venv sit-package's dir` got changed,  the package A of `new_venv` will also change accordingly.